app/services/auth.js

```diff
@@ -1,276 +1,196 @@
-    return state === 'signed-out';
-  }),
-
-  signingIn: computed('state', function () {
-    let state = this.state;
-    return state === 'signing-in';
-  }),
-
-  storeData(data, storage) {
-    if (data.token) {
-      storage.setItem('travis.token', data.token);
+  validateUserData(user) {
+    const hasChannelsOnPro = field => field === 'channels' && this.isProVersion;
+    const hasAllFields = USER_FIELDS.every(field => isPresent(user[field]) || hasChannelsOnPro(field));
+    if (!hasAllFields || !user.correct_scopes) {
+      throw new Error('User validation failed');
     }
-    return storage.setItem('travis.user', JSON.stringify(data.user));
-  },
-
-  loadUser(user) {
-    let store = this.store,
-      userClass = store.modelFor('user'),
-      serializer = store.serializerFor('user'),
-      normalized = serializer.normalizeResponse(store, userClass, user, null, 'findRecord');
-
-    store.push(normalized);
-    const record =  store.recordForId('user', user.id);
-    const installation = store.peekAll('installation').findBy('owner.id', user.id) || null;
-    record.setProperties({ installation });
-    return record;
   },
 
-  expectedOrigin() {
-    let endpoint = this.endpoint;
-    if (endpoint && endpoint[0] === '/') {
-      return this.receivingEnd;
-    } else {
-      let matches = endpoint.match(/^https?:\/\/[^\/]*/);
-      if (matches && matches.length) {
-        return matches[0];
-      }
+  reportToIntercom() {
+    if (this.isProVersion && intercom.enabled) {
+      const { id, name, email, firstLoggedInAt, secureUserHash } = this.currentUser;
+      this.intercom.setProperties({
+        'user.id': id,
+        'user.name': name,
+        'user.email': email,
+        'user.createdAt': firstLoggedInAt,
+        'user.hash': secureUserHash
+      });
     }
   },
 
+  // refreshUserData(user, include = []) {
+  //   includes = includes.concat(include, ['owner.installation']).uniq();
+  //   if (!user) {
+  //     let data = this.userDataFromStorage();
+  //     if (data) {
+  //       user = data.user;
+  //     }
+  //   }
+  //   if (user) {
+  //     return this.ajax.get(`/users/${user.id}`)
+  //       .then(data => {
+  //         let userRecord;
+  //         if (data.user.correct_scopes) {
+  //           userRecord = this.loadUser(data.user);
+  //           userRecord.get('permissions');
+  //           if (this.signedIn) {
+  //             data.user.token = user.token;
+  //             Travis.trigger('user:refreshed', data.user);
+  //           }
+  //           return this.store.queryRecord('user', {
+  //             current: true,
+  //             included: includes.join(',')
+  //           });
+  //         } else {
+  //           return Promise.reject();
+  //         }
+  //       })
+  //       .then(({ installation = null }) => {
+  //         this.currentUser.setProperties({ installation });
+  //       });
+  //   } else {
+  //     return Promise.resolve();
+  //   }
+  // },
```

Comment:
Comment needed?

Reply:
Yeah, this method is quite complex and I kept it for reference while fixing tests. I'll make sure to clean up before merge 👍